pkg/apiserver/handlers.go

```diff
@@ -139,6 +139,13 @@ func tooManyRequests(req *http.Request, w http.ResponseWriter) {
 func RecoverPanics(handler http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
 		defer runtime.HandleCrash(func(err interface{}) {
+			if e, ok := err.(error); ok {
```

Comment:
> I think there are only 2 things that can panic in the current timeout handler: a call to t.handler.ServerHTTP, and tw.timeout.

You are right. See my last reply (we commented at the same time). Still, I feel the current approach is slightly better. But I have no strong opinion.